### DISCUSSION TITLE
Stability and real usage

### DISCUSSION THREAD

**Discussion by conormuirhead** [ThinkUniform/uniformcss#13] [General]
Hey @jinsupark, I've been eyeing Uniform CSS over the last several weeks and playing with it a bit too. I really appreciated the configurability that you've incorporated, and they way you've made set it up to be expandable — thanks for doing the hard work to build this whole thing!

I'm considering incorporating it into the app where I work, but wanted to get a sense of two things from you:

### Stability
- how stable do you think Uniform CSS is?

### Real-world usage 
- What kind of real-world usage has it received?
- Do you all use this at your day job?
- Have you used it in a web app context, or just for marketing sites?

If you'd be willing, I'd love to hop on a call with you sometime to discuss some details and connect. Would you be willing to do something like that?

Thanks!
-Conor

**Comment by jinsupark** [comment]
Hi @conormuirhead thank you for checking out this project!

**Stability**

Although not as popular and well known as some of the older frameworks like Tachyon and Tailwind CSS, due to the simplistic and predictable nature of Uniform CSS, I would say it is quite stable where I feel confident that it can be used in production. 

In the CSS world, I think calling something a "CSS Framework" comes with it a certain level of weight and technical debt in people's minds. Something is either an extremely lightweight grid library that's under 200 lines of code or something that is deeply nested like Bootstrap or Foundation that requires a considerable amount of buy-in and proven stability. 

With that being said, even though Uniform CSS is advertised as a CSS Framework, I see this project more as a _complementary_ library to help developers quickly customize and generate utility classes to allow them to be more productive in their workflow. Of course, it can be utilized in a holistic manner, however, due to the simplistic nature of utility classes, I don't believe it is something that requires the same level of stability validation as some of the more opinionated complex projects out there.

The codebase is also reasonably straightforward and there aren't any black-box complexities. This makes the process of debugging, customizing and extending very easy to do.

**Real-world usage**

We use this every day at work for both prototyping purposes and production. We are currently fully invested in using Uniform CSS for both our marketing site and the front-end UI for our core product. We were already using Sass as part of our front-end stack and therefore integrating Uniform CSS was a very smooth process since it's a Sass module.

The current build is serving us pretty well and fulfills all our needs so far. I am devoted to continuously improving and developing the project further.

I look forward to speak with you!
